src/server/shell/accessibility_manager.cpp

```diff
@@ -42,12 +52,81 @@ void mir::shell::AccessibilityManager::repeat_delay(int new_delay) {
     repeat_delay_ = new_delay;
 }
 
-void mir::shell::AccessibilityManager::override_key_repeat_settings(bool enable)
-{
-    enable_key_repeat = enable;
-}
-
 void mir::shell::AccessibilityManager::notify_helpers() const {
     for(auto const& helper: keyboard_helpers)
         helper->repeat_info_changed(repeat_rate(), repeat_delay());
 }
+
+struct MouseKeysTransformer: public mir::input::InputEventTransformer::Transformer
+{
+    bool transform_input_event(
+        mir::input::InputEventTransformer::EventDispatcher const& dispatcher,
+        mir::input::EventBuilder* builder,
+        MirEvent const& event) override
+    {
+        using namespace mir; // For operator<<
+
+        std::stringstream ss;
+        ss << const_cast<MirEvent&>(event);
+        mir::log_debug("%s", ss.str().c_str());
```

Comment:
Does there not just exist a "toString()" method here to avoid creating the string stream? If not, can we add one? e.g.

```suggestion
        mir::log_debug("%s", event.toString());
```

Reply:
The only thing I could find was `operator<<` in `event_printer.h`. I guess since this is mostly debug code that it doesn't matter if its slightly messy. What do you think?

Reply:
Eh I don't really care, but it is just another object to create every time an input event happens, which could get a bit costly.

Reply:
We'll it is debug code after all. Anyway, I removed these few lines in the latest batch of commits